File: DevExpress-Demo/VB/SpellCheckerDemo.Wpf/DemoTesting/Fixtures.vb
Imports DevExpress.Xpf.DemoBase.DemoTesting

Namespace SpellCheckerDemo.Tests

    Public Class SpellCheckerDemoModuleAccessor
        Inherits DemoModulesAccessor(Of SpellCheckerDemoModule)

        Public Sub New(ByVal fixture As BaseDemoTestingFixture)
            MyBase.New(fixture)
        End Sub
    End Class

    Public MustInherit Class BaseSpellCheckerTestingFixture
        Inherits BaseDemoTestingFixture

        Private ReadOnly modulesAccessor As SpellCheckerDemoModuleAccessor

        Public Sub New()
            modulesAccessor = New SpellCheckerDemoModuleAccessor(Me)
        End Sub
    End Class

    Public Class SpellCheckerCheckAllDemosFixture
        Inherits CheckAllDemosFixture

    End Class
End Namespace

File: DevExpress-Demo/CS/SpellCheckerDemo.Wpf/DemoTesting/Fixtures.cs
using DevExpress.Xpf.DemoBase.DemoTesting;

namespace SpellCheckerDemo.Tests {
    public class SpellCheckerDemoModuleAccessor : DemoModulesAccessor<SpellCheckerDemoModule> {
        public SpellCheckerDemoModuleAccessor(BaseDemoTestingFixture fixture)
            : base(fixture) {
        }
    }
    public abstract class BaseSpellCheckerTestingFixture : BaseDemoTestingFixture {
        readonly SpellCheckerDemoModuleAccessor modulesAccessor;
        public BaseSpellCheckerTestingFixture() {
            this.modulesAccessor = new SpellCheckerDemoModuleAccessor(this);
        }
    }
    public class SpellCheckerCheckAllDemosFixture : CheckAllDemosFixture {
    }
}
